setup.py

```diff
@@ -780,76 +826,135 @@ def get_provider_package_from_package_id(package_id: str):
-    those extras are installed.
+    In case of regular installation (providers installed from packages), we should add extra dependencies to
+    Airflow - to get the providers automatically installed when those extras are installed.
+
+    For providers installed from sources we skip that step. That helps to test and install airflow with
+    all packages in CI - for example when new providers are added, otherwise the installation would fail
+    as the new provider is not yet in PyPI.
 
     """
     for provider in ALL_PROVIDERS:
-        add_provider_packages_to_requirements(provider, [provider])
-    add_provider_packages_to_requirements("all", ALL_PROVIDERS)
-    add_provider_packages_to_requirements("devel_ci", ALL_PROVIDERS)
-    add_provider_packages_to_requirements("devel_all", ALL_PROVIDERS)
-    add_provider_packages_to_requirements("all_dbs", ALL_DB_PROVIDERS)
-    add_provider_packages_to_requirements("devel_hadoop", ["apache.hdfs", "apache.hive", "presto"])
+        add_provider_packages_to_extras_requirements(provider, [provider])
+    add_provider_packages_to_extras_requirements("all", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_ci", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_all", ALL_PROVIDERS)
+    add_provider_packages_to_extras_requirements("all_dbs", ALL_DB_PROVIDERS)
+    add_provider_packages_to_extras_requirements("devel_hadoop", ["apache.hdfs", "apache.hive", "presto"])
+
+
+class Develop(develop_orig):
+    """Forces removal of providers in editable mode."""
+
+    def run(self):
+        self.announce('Installing in editable mode. Uninstalling provider packages!', level=log.INFO)
+        # We need to run "python3 -m pip" because it might be that older PIP binary is in the path
+        # And it results with an error when running pip directly (cannot import pip module)
+        # also PIP does not have a stable API so we have to run subprocesses ¯\_(ツ)_/¯
+        try:
+            installed_packages = (
+                subprocess.check_output(["python3", "-m", "pip", "freeze"]).decode().splitlines()
+            )
+            airflow_provider_packages = [
+                package_line.split("=")[0]
+                for package_line in installed_packages
+                if package_line.startswith("apache-airflow-providers")
+            ]
+            self.announce(f'Uninstalling ${airflow_provider_packages}!', level=log.INFO)
+            subprocess.check_call(["python3", "-m", "pip", "uninstall", "--yes", *airflow_provider_packages])
+        except subprocess.CalledProcessError as e:
+            self.announce(f'Error when uninstalling airflow provider packages: {e}!', level=log.WARN)
+        super().run()
+
+
+class Install(install_orig):
+    """Forces installation of providers from sources in editable mode."""
 
+    def run(self):
+        self.announce('Standard installation. Providers are installed from packages', level=log.INFO)
```

Comment:
This would show up to any user who installs an sdist, so I don't think we should show this here.

Reply:
It's only shown when `-vv` option is used with PIP I believe. This is a known problem that PIP is not really useful if you want to print any warning or information while installing because by default it only prints what authors of PIP wanted. I will double check it and remove (but I think I checked it before and it only showed up with high verbosity level).

Reply:
Ah if it only shows up under v>=2 that's alright then.

Reply:
Actually it shows up even with one -v, but the output of -v is fantastically (or rather wanted to say fanatically) verbose in this case, so it does not really matter. This is about 5% of the log printed with -v :)

```
copying airflow/www/templates/airflow/variable_edit.html -> build/lib/airflow/www/templates/airflow
  copying airflow/www/templates/airflow/variable_list.html -> build/lib/airflow/www/templates/airflow
  copying airflow/www/templates/airflow/xcom.html -> build/lib/airflow/www/templates/airflow
  creating build/lib/airflow/www/templates/analytics
  copying airflow/www/templates/analytics/google_analytics.html -> build/lib/airflow/www/templates/analytics
  copying airflow/www/templates/analytics/metarouter.html -> build/lib/airflow/www/templates/analytics
  copying airflow/www/templates/analytics/segment.html -> build/lib/airflow/www/templates/analytics
  creating build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/custom_icons.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/dag_docs.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/flash.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/index.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/loading_dots.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/navbar.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/navbar_menu.html -> build/lib/airflow/www/templates/appbuilder
  copying airflow/www/templates/appbuilder/navbar_right.html -> build/lib/airflow/www/templates/appbuilder
  creating build/lib/airflow/api_connexion/openapi
  copying airflow/api_connexion/openapi/v1.yaml -> build/lib/airflow/api_connexion/openapi
  installing to build/bdist.linux-x86_64/wheel
  running install
  Standard installation. Providers are installed from packages
  running install_lib
  creating build/bdist.linux-x86_64
  creating build/bdist.linux-x86_64/wheel
  creating build/bdist.linux-x86_64/wheel/airflow
  creating build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/schema.json -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/json_schema.py -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/helpers.py -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/enums.py -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/serialized_objects.py -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/serialization/__init__.py -> build/bdist.linux-x86_64/wheel/airflow/serialization
  copying build/lib/airflow/customized_form_field_behaviours.schema.json -> build/bdist.linux-x86_64/wheel/airflow
  creating build/bdist.linux-x86_64/wheel/airflow/macros
  copying build/lib/airflow/macros/hive.py -> build/bdist.linux-x86_64/wheel/airflow/macros
  copying build/lib/airflow/macros/__init__.py -> build/bdist.linux-x86_64/wheel/airflow/macros
  creating build/bdist.linux-x86_64/wheel/airflow/mypy
  creating build/bdist.linux-x86_64/wheel/airflow/mypy/plugin
  copying build/lib/airflow/mypy/plugin/decorators.py -> build/bdist.linux-x86_64/wheel/airflow/mypy/plugin

```